### PR TITLE
Task 47

### PR DESCRIPTION
**Pull request by paulvxx**
Created separate GitHub actions to  test,  build,  and publish on release.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Build and Publish
 
 on:
   push:
-    branches: [ "main" ] 
+    branches: [ "task_47" ]
```
Comment (seanbsu) [inline .github/workflows/build.yml:5]: shouldnt this be for on push to main? also is there a reason you took out the other python version tests?

**Review by seanbsu — CHANGES_REQUESTED** [review]
left a comment on build.yml but just to be safe. you have it on push to your branch not to main it should be on push to main for CI. Also was there a reason to take out testing on other python versions?

File: .github/workflows/build.yml (hunk 1 above)
Comment (paulvxx) [reply .github/workflows/build.yml:5]: True.

**Review by seanbsu — CHANGES_REQUESTED** [review]
ok ojne last change request. we should make sure the lint tests different python versions again cause right now it only tests 3.8. somwhere along the line the lint.sh file got changed. the matrix was orginially python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]

**Review by seanbsu — APPROVED** [review]

**Comment by paulvxx** [comment]
- Fixed some bugs
- Minor errors

[t:30]

**Pull request merged**